src/packet/key/secret.rs

```diff
@@ -451,24 +511,48 @@ impl PublicKeyTrait for SecretSubkey {
 }
 
 impl<D: PublicKeyTrait + crate::ser::Serialize> PublicKeyTrait for SecretKeyInner<D> {
-    fn verify_signature(&self, hash: HashAlgorithm, hashed: &[u8], sig: &[Mpi]) -> Result<()> {
+    fn verify_signature(
+        &self,
+        hash: HashAlgorithm,
+        hashed: &[u8],
+        sig: &SignatureBytes,
+    ) -> Result<()> {
         self.details.verify_signature(hash, hashed, sig)
     }
 
-    fn encrypt<R: rand::Rng + rand::CryptoRng>(
-        &self,
-        rng: &mut R,
-        plain: &[u8],
-    ) -> Result<Vec<Mpi>> {
+    fn encrypt<R: rand::Rng + rand::CryptoRng>(&self, rng: R, plain: &[u8]) -> Result<Vec<Mpi>> {
         self.details.encrypt(rng, plain)
     }
 
     fn serialize_for_hashing(&self, writer: &mut impl std::io::Write) -> Result<()> {
+        use byteorder::{BigEndian, WriteBytesExt};
+
+        // FIXME: avoid duplication with key/public.rs!
```

Comment:
still want to do this?

Reply:
oh right. i'd like to, but at some point confused myself into being unsure if it's easily possible.

Reply:
yay, that's less annoying. 80a8a2d691de5b457c5ad9e8809828de22896cc9